minsar/additions/mintpy/auto_path.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+############################################################
+# Program is part of MintPy                                #
+# Copyright (c) 2013, Zhang Yunjun, Heresh Fattahi         #
+# Author: Zhang Yunjun, Mar 2018                           #
+############################################################
+
+
+import os
+import re
+import glob
+import numpy as np
+
+# Auto setting for file structure of Univ. of Miami, as shown below.
+# It required 3 conditions: 1) autoPath = True
+#                           2) $SCRATCHDIR is defined in environmental variable
+#                           3) input custom template with basename same as project_name
+# Change it to False if you are not using it.
+autoPath = True
+
+
+# Default path of data files from different InSAR processors to be loaded into MintPy
+isceAutoPath = '''##----------Default file path of ISCE-topsStack products
+mintpy.load.processor      = isce
+mintpy.load.metaFile       = ${PROJECT_DIR}/master/IW*.xml
+mintpy.load.baselineDir    = ${PROJECT_DIR}/baselines
+
+mintpy.load.unwFile        = ${PROJECT_DIR}/merged/interferograms/*/filt*.unw
+mintpy.load.corFile        = ${PROJECT_DIR}/merged/interferograms/*/filt*.cor
+mintpy.load.connCompFile   = ${PROJECT_DIR}/merged/interferograms/*/filt*.unw.conncomp
+mintpy.load.ionoFile       = None
+mintpy.load.intFile        = None
+
+mintpy.load.demFile        = ${PROJECT_DIR}/merged/geom_master/hgt.rdr
+mintpy.load.lookupYFile    = ${PROJECT_DIR}/merged/geom_master/lat.rdr
+mintpy.load.lookupXFile    = ${PROJECT_DIR}/merged/geom_master/lon.rdr
+mintpy.load.incAngleFile   = ${PROJECT_DIR}/merged/geom_master/los.rdr
+mintpy.load.azAngleFile    = ${PROJECT_DIR}/merged/geom_master/los.rdr
+mintpy.load.shadowMaskFile = ${PROJECT_DIR}/merged/geom_master/shadowMask.rdr
+mintpy.load.bperpFile      = None
+'''
+iscestripmapAutoPath = '''##----------Default file path of ISCE-topsStack products
+mintpy.load.processor      = isce
+mintpy.load.metaFile       = ${PROJECT_DIR}/masterShelve/data.dat
```

Comment:
The path for metaFile is not correct

Reply:
I am making a copy of masterShelve in the master directory. SOmethings to be consistent with topsstack where the info is from ${PROJECT_DIR}/master/IW1.xml.

Reply:
@mirzaees Is there a way to get the master date info into smallBaseline_wrapper. At that point, inputs dictionary has master date as 'None', so, I am directing it to the first date/folder in SLC directory. Can the masterdate info be updated into "inps"

Reply:
@bhuvankumaru I see, thank you
It is not really important to give it the master image as the masterShelve in all SLC folders are the same
smallbaselineWrapper.py can read the master from template file.
generally it is not a good idea to detach from the main repo very much, I have made a PR to mintpy, let's see if Yunjun does not approve, we can temporarily add this